extensions/obviousAlexC/penPlus.js

```diff
@@ -0,0 +1,2621 @@
+              y = Math.floor(y - 1);
+              const colorIndex = (y * curCostume.width + x) * 4;
+              if (textureData[colorIndex] && x < curCostume.width && x >= 0) {
+                return colors.rgbtoSColor({
+                  R: textureData[colorIndex] / 2.55,
+                  G: textureData[colorIndex + 1] / 2.55,
+                  B: textureData[colorIndex + 2] / 2.55,
+                });
+              }
+              return colors.rgbtoSColor({ R: 100, G: 100, B: 100 });
+            }
+          }
+        }
+      )
+      .addArgument("x", 1)
+      .addArgument("y", 1)
+      .addArgument("costume", null, null, "penPlusCostumes");
+
+    extension
+      .addBlock(
+        "get data uri of [costume] in the pen+ costume library",
+        "getPenPlusCostumeURI",
+        Scratch.BlockType.REPORTER,
+        ({ costume }) => {
+          const curCostume = penPlusCostumeLibrary[costume];
+          if (curCostume) {
+            const textureData = textureFunctions.getTextureAsURI(
+              curCostume.texture,
+              curCostume.width,
+              curCostume.height
+            );
+            if (textureData) {
+              return textureData;
+            }
+            return "";
+          }
+        }
+      )
+      .addArgument("costume", null, null, "penPlusCostumes");
+
+    extension.addLabel("Advanced options");
+
+    extension
+      .addBlock(
+        "turn advanced setting [Setting] [onOrOff]",
+        "turnAdvancedSettingOff",
+        Scratch.BlockType.COMMAND,
+        ({ Setting, onOrOff }) => {
+          if (onOrOff == "on") {
+            penPlusAdvancedSettings[Setting] = true;
+            return;
+          }
+          penPlusAdvancedSettings[Setting] = false;
+        }
+      )
+      .addArgument("Setting", null, null, "advancedSettingsMenu")
+      .addArgument("onOrOff", null, null, "onOffMenu");
+
+    extension
+      .addBlock(
+        "set [setting] to [value]",
+        "setAdvancedOptionValueTo",
+        Scratch.BlockType.COMMAND,
+        ({ setting, value }) => {
+          switch (setting) {
+            case "depthMax":
+              penPlusAdvancedSettings._maxDepth = Math.max(value, 100);
+              break;
+
+            default:
+              break;
+          }
+        }
+      )
+      .addArgument("setting", null, null, "advancedSettingValuesMenu")
+      .addArgument("value", "1000");
+  }
+
+  //? A small hack to stop the renderer from immediatly dying. And to allow for immediate use
+  {
+    if (!Scratch.vm.renderer._penSkinId) {
+      window.vm.renderer.createPenSkin();
+    }
+    renderer.penClear(Scratch.vm.renderer._penSkinId);
+    Scratch.vm.renderer.penLine(
+      Scratch.vm.renderer._penSkinId,
+      {
+        color4f: [0, 0, 1, 1],
+        diameter: 1,
+      },
+      0,
+      0,
+      0,
+      0
+    );
+
+    penPlusShaders.pen.program = shaderManager._shaderCache.line[0].program;
```

Comment:
`Property 'program' does not exist on type 'ProgramInfo[]'.`

Reply:
Where are you getting this error exactly. Because in practice the error never showed up in my testing. Guessing its a type error because of the way its phrased but the property does since that is where the webGL program is stored.